cpp/src/strings/split/split_record.cu

```diff
@@ -23,123 +25,56 @@
-      return string_index_pair{d_str.data() + byte_offset,
-                               d_str.byte_offset(end_pos) - byte_offset};
-    }
+template <typename Tokenizer>
+std::unique_ptr<column> split_record_fn(strings_column_view const& input,
+                                        Tokenizer tokenizer,
+                                        rmm::cuda_stream_view stream,
+                                        rmm::mr::device_memory_resource* mr)
+{
+  if (input.is_empty()) { return make_empty_column(type_id::LIST); }
+  if (input.size() == input.null_count()) {
+    auto offsets = std::make_unique<column>(input.offsets(), stream, mr);
+    auto results = std::make_unique<column>(input.parent(), stream, mr);
+    return make_lists_column(input.size(),
+                             std::move(offsets),
+                             std::move(results),
+                             input.null_count(),
+                             copy_bitmask(input.parent(), stream, mr),
+                             stream,
+                             mr);
   }
 
-  __device__ void operator()(size_type idx)
-  {
-    if (d_strings.is_null(idx)) { return; }
+  // builds the offsets and the vector of all tokens
+  auto [offsets, tokens] = split_helper(input, tokenizer, stream, mr);
 
-    auto const token_offset = d_token_offsets[idx];
-    auto const token_count  = d_token_offsets[idx + 1] - token_offset;
-    auto d_result           = d_tokens + token_offset;
-    auto const d_str        = d_strings.element<string_view>(idx);
-    if (d_str.empty()) {
-      // Pandas str.split("") for non-whitespace delimiter is an empty string
-      *d_result = string_index_pair{"", 0};
-      return;
-    }
+  // build a strings column from the tokens
+  auto strings_child = make_strings_column(tokens.begin(), tokens.end(), stream, mr);
 
-    size_type token_idx = 0;
-    size_type start_pos = 0;               // updates only if moving forward
-    size_type end_pos   = d_str.length();  // updates only if moving backward
-    while (token_idx < token_count - 1) {
-      auto const delimiter_pos = dir == Dir::FORWARD ? d_str.find(d_delimiter, start_pos)
-                                                     : d_str.rfind(d_delimiter, start_pos, end_pos);
-      if (delimiter_pos == string_view::npos) break;
-      auto const token = resolve_token(d_str, start_pos, end_pos, delimiter_pos);
-      if (dir == Dir::FORWARD) {
-        d_result[token_idx] = token;
-        start_pos           = delimiter_pos + d_delimiter.length();
-      } else {
-        d_result[token_count - 1 - token_idx] = token;
-        end_pos                               = delimiter_pos;
-      }
-      token_idx++;
-    }
+  return make_lists_column(input.size(),
+                           std::move(offsets),
+                           std::move(strings_child),
+                           input.null_count(),
+                           copy_bitmask(input.parent(), stream, mr),
```

Comment:
Again, making a lists column with bitmask copied from the input. I'm not sure if this will generate non-empty nulls.